packages/bundles-static/src/components/entry-point/entry-point.js

```diff
@@ -22,78 +19,50 @@ const AsyncApplicationRoutes = React.lazy(() =>
-      // Your custom configuration, according to the Apollo cache documentation.
-      // https://www.apollographql.com/docs/react/caching/cache-configuration/
-    },
-  });
+let headers = {
+  Accept: 'application/json',
+}
 
-  const client = new ApolloClient({
-    link: ApolloLink.from([restLink, apolloClient.link]),
-    cache: new InMemoryCache(),
-  });
+if ( window.app.env === 'development' )
+  headers.Authorization = `Bearer ${window.sessionStorage.getItem('sessionToken')}`
 
-  return (
-    <ApolloProvider client={client}>
-      <Switch>
-        {
-          /* For development, it's useful to redirect to the actual
-          application routes when you open the browser at http://localhost:3001 */
-          process.env.NODE_ENV === 'production' ? null : (
-            <Redirect
-              exact={true}
-              from="/:projectKey"
-              to={`/:projectKey/${ROOT_PATH}`}
-            />
-          )
-        }
-        <Route
-          path={`/:projectKey/${ROOT_PATH}`}
-          component={AsyncApplicationRoutes}
-        />
-        {/* Catch-all route */}
-        <RouteCatchAll />
-      </Switch>
-    </ApolloProvider>
-  );
-};
-ApplicationBundleManager.displayName = 'ApplicationBundleManager';
+const restLink = new RestLink({
+  uri: `${mcApiUrl}/proxy/ctp/${projectKey}`,
+  headers: headers,
+  credentials: 'include',
+});
 
-// Ensure to setup the global error listener before any React component renders
-// in order to catch possible errors on rendering/mounting.
-setupGlobalErrorListener();
+const apolloClient = createApolloClient({
+  cache: {
+    // Your custom configuration, according to the Apollo cache documentation.
+    // https://www.apollographql.com/docs/react/caching/cache-configuration/
+  },
+});
 
-class EntryPoint extends React.Component {
-  static displayName = 'EntryPoint';
-  render() {
-    return (
-      <ApplicationShell
-        environment={window.app}
-        onRegisterErrorListeners={({ dispatch }) => {
-          Sdk.Get.errorHandler = (error) =>
-            globalActions.handleActionError(error, 'sdk')(dispatch);
-        }}
-        applicationMessages={loadMessages}
-        DEV_ONLY__loadNavbarMenuConfig={() =>
-          import('../../../menu.json').then((data) => data.default || data)
-        }
-        render={() => <ApplicationBundleManager />}
-      />
-    );
-  }
-}
+const client = new ApolloClient({
+  link: ApolloLink.from([restLink, apolloClient.link]),
+  cache: new InMemoryCache(),
+});
+
+
+const EntryPoint = () => (
+    <ApplicationShell environment={window.app} applicationMessages={loadMessages} apolloClient={client}
+                      onRegisterErrorListeners={({ dispatch }) => {
+                          Sdk.Get.errorHandler = (error) =>
+                            globalActions.handleActionError(error, 'sdk')(dispatch);
+                            }
+                      }>
+      <AsyncApplicationRoutes />
+    </ApplicationShell>
+);
```

Comment:
Everything in this file looks the same as in the dynamic module. Maybe we could pull out common things into super class of the bundles-core?

Reply:
We cannot pull the entire entrypoint component. But yes, as I already added TODO in the file about refactoring. we can do this later.